Fix calculated fields doesn't handle numeric values with non US locale

attrib opened the pull request:
Fixes https://github.com/salesagility/SuiteCRM/issues/5511

When using non US locale and using a mathematical caluclation it returns the string in the wrong number format, which on save will result in incorrect calculations.

Example calc:

`multiply({P0};{P1})`

if P0 is 0.8 and P1 is 0.5 the calculated field will set the string `"0.4" ` to the selected field. If this is a numeric field on save it will parse the result. In german local the dot is the thousand separator, resulting the field beeing saved as 4 and not 0.4 in the database.

<!--- Provide a general summary of your changes in the Title above -->
<!--- Please be aware that as of the 31st January 2022 we no longer support 7.10.x.
New PRs to hotfix-7.10.x will be invalid. If your fix is still applicable to 7.12.x, 
please create the pull request to the hotfix branch accordingly. -->


## Description
<!--- Describe your changes in detail -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here unless your commit contains the issue number -->
<!--- Ensure that all code ``` is surrounded ``` by triple back quotes. This can also be done over multiple lines -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How To Test This
<!--- Please describe in detail how to test your changes. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
<!--- Go over all the following points and check all the boxes that apply. --->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! --->
- [ ] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

<!--- Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation. --->

<!--- If it reports back that there are problems, you can log into the Travis system and check the log report for your pull request to see what the problem was. --->

attrib committed:
Fix #5511 -  Calculated fields doesn't handle numeric values with non US locale

When using non US locale and using a mathematical caluclation it returns the string in the wrong number format, which on save will result in incorrect calculations.

Example calc:

`multiply({P0};{P1})`

if P0 is 0.8 and P1 is 0.5 the calculated field will set the string `"0.4" ` to the selected field. If this is a numeric field on save it will parse the result. In german local the dot is the thousand separator, resulting the field beeing saved as 4 and not 0.4 in the database.

serhiisamko091184 commented:
Hello, @attrib 

thank you for the contribution,

Could you, please, update your commit message following our documentation (add number of the issue to your message): https://docs.suitecrm.com/community/contributing-code/bugs/

As an example:
![image](https://github.com/salesagility/SuiteCRM/assets/116086008/bbf81e7f-59d8-4457-b8e0-d9082fd3f563)

Thanks a lot i advance!

Regards,
Serhii

attrib commented:
Rebased and changed my old solution.

My old solution run into issues, if working with numbers > 9999, as it introcuded the thousand seperator, which broke my old logic.

SinergiaCRM commented:
We have added this code (https://github.com/SinergiaTIC/SinergiaCRM/pull/49) but contemplating more variables because this part only includes these types of number: 'float', 'decimal', 'currency' and 'int'. We think that this have to search other possibilities of numbers, since Studio doesn't have it, if it were added with code.
